src/components/settings/settings/EditSettingsForm.tsx

```diff
@@ -354,14 +356,24 @@ class EditSettingsForm extends Component<IProps, IState> {
     this.props.form.submit({
       onSuccess: (form: Form) => {
         const values = form.values();
-        const { accentColor } = values;
+        const { accentColor, twoFactorAutoCatcher } = values;
+
         if (accentColor.trim().length === 0) {
           values.accentColor = DEFAULT_ACCENT_COLOR;
         }
         const { progressbarAccentColor } = values;
         if (progressbarAccentColor.trim().length === 0) {
           values.progressbarAccentColor = DEFAULT_ACCENT_COLOR;
         }
+
+        // If twoFactorAutoCatcher is enabled, but the string input is empty, set it to the default value
+        if (
```

Comment:
By this code, it looks like the array will always have a value? But, if I, as a security conscious user, want to turn this off, then shouldn't the array remain empty?

Reply:
1. yes
2. yes
maybe we get if the toggle gets activated then the array get filled?

Reply:
yes, if the toggle gets activated, the value of the string can be non-empty
If the toggle gets deactivated, the value of the string can become empty string